openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -101,37 +102,47 @@ class AccessTokenView(RatelimitMixin, _DispatchingView):
     def dispatch(self, request, *args, **kwargs):
         response = super(AccessTokenView, self).dispatch(request, *args, **kwargs)
 
-        if response.status_code == 200 and request.POST.get('token_type', '').lower() == 'jwt':
-            client_id = self._get_client_id(request)
-            adapter = self.get_adapter(request)
-            expires_in, scopes, user = self._decompose_access_token_response(adapter, response)
-            issuer, secret, audience, filters, is_client_restricted = self._get_client_specific_claims(
-                client_id,
-                adapter
-            )
-
-            content = {
-                'access_token': JwtBuilder(
-                    user,
-                    secret=secret,
-                    issuer=issuer,
-                ).build_token(
-                    scopes,
-                    expires_in,
-                    aud=audience,
-                    additional_claims={
-                        'filters': filters,
-                        'is_restricted': is_client_restricted,
-                    },
-                ),
-                'expires_in': expires_in,
-                'token_type': 'JWT',
-                'scope': ' '.join(scopes),
-            }
-            response.content = json.dumps(content)
+        if response.status_code == 200:
+            monitoring_utils.set_custom_metric('oauth_grant_type', request.POST.get('grant_type', ''))
+
+            if request.POST.get('token_type', '').lower() == 'jwt':
+                response.content = self._build_jwt_response_from_access_token_response(request, response)
+
+                monitoring_utils.set_custom_metric('oauth_token_type', 'jwt')
+            else:
+                monitoring_utils.set_custom_metric('oauth_token_type', 'oauth')
 
         return response
 
+    def _build_jwt_response_from_access_token_response(self, request, response):
```

Comment:
FYI: This is basically the original code moved out to its own method.  There are further refactors possible, but not in this PR. :)